Potions now work.

Cranite opened the pull request:
ItemExchange will now support the exchange of potions without the need for systemic changes or upgrades.

Notes:

1) The display name will technically be stored twice in a potion's exchange rule data, but as AdditionalMetadata classes only receive the item's metadata, it's impossible to receive the already parsed display name without further changes.

2) The display info will not output the specifics of a potion's effects. It can be added back if necessary, but no other kind of metadata displays such intimate information, except perhaps books where knowing the author and title is necessary.

3) The hard coded potions names are necessary as ItemExchange doesn't use CivModCore and so doesn't use various APIs to deduce a vanilla potion's name by its base effect, at least not without systemic changes outside the scope of this update.

4) Tested with vanilla potions and potions resulting from brewing.

Commit:
Potions now work.

ItemExchange will not support the exchange of potion without the need for systemic changes or upgrades.

Commit:
Oopsie

Missed out the breaks, the code is copied from MinecraftUtilities where the names would have been returned, but as the names are being assigned there was nothing breaking out of the switch so every vanilla potion was named Potion of Luckk. Oops.

Gjum commented:
Does this work for [brews](https://github.com/DevotedMC/Brewery) too?

How difficult will it be to update this to 1.13/14 down the line?

ghost commented:
It works with brews :) as said in note four. As for upgrades, unless potion data has been changed wildly, it would only require minor modifications to serialisation/deserialisation.

Maxopoly commented on src/main/java/com/untamedears/ItemExchange/metadata/PotionMetadata.java:
```diff
@@ -3,118 +3,206 @@
+					PotionEffectType type = PotionEffectType.getById(effectId);
+					metadata.effects.add(new PotionEffect(type, duration, amplifier, ambient));
+				}
+				catch (Exception ignored) {}
```
Might be good to print and not just silently swallow this exception so actual issues with it can be found and fixed in the future

Cranite replied:
Done :) I also added logging to the other parser

Maxopoly commented:
Would probably be good to eventually centralize potion naming in CivModCore. Until then (if ever) this is good, decent code.

Commit:
Added logging.

ItemExchange will now output to the console if errors occurred while parsing a potion's effects.

ghost commented:
The potion name switch is pulled from my [MinemanUtilities](https://github.com/minemanmods/MinemanUtilities/blob/master/src/main/java/com/github/minemanmods/MinemanUtilities/PotionAPI.java#L109) library. It might be worth just copying the most of it.
